packages/plugins/src/hardhat/tasks.ts

```diff
@@ -1352,3 +1355,171 @@ export const listProjectsTask = async ({}, hre: HardhatRuntimeEnvironment) => {
+
+const claimOwnershipTask = async (
+  args: {
+    configPath: string
+    referenceName: string
+    silent: boolean
+  },
+  hre
+) => {
+  const { configPath, referenceName, silent } = args
+  const provider = hre.ethers.provider
+  const signer = provider.getSigner()
+
+  const spinner = ora({ isSilent: silent })
+  spinner.start('Checking project registration...')
+
+  const parsedConfig = loadParsedChugSplashConfig(configPath)
+
+  // Throw an error if the project has not been registered
+  if (
+    (await isProjectRegistered(signer, parsedConfig.options.projectName)) ===
+    false
+  ) {
+    errorProjectNotRegistered(
+      await getChainId(hre.ethers.provider),
+      hre.network.name,
+      configPath
+    )
+  }
+
+  spinner.start('Claiming proxy ownership...')
+
+  const manager = await getChugSplashManager(
+    signer,
+    parsedConfig.options.projectName
+  )
+
+  const signerAddress = await signer.getAddress()
+  await (
+    await manager.transferProxyOwnership(referenceName, signerAddress)
+  ).wait()
+
+  spinner.succeed(`Proxy ownership claimed by address ${signerAddress}`)
+}
+
+task(TASK_CHUGSPLASH_CLAIM_OWNERSHIP)
+  .addPositionalParam(
+    'configPath',
+    'Path to the ChugSplash config file for the project that owns the target contract'
+  )
+  .addPositionalParam(
+    'referenceName',
+    'Reference name of the contract that should be transferred to you'
+  )
+  .addFlag('silent', "Hide all of ChugSplash's output")
+  .setAction(claimOwnershipTask)
+
+const transferOwnershipTask = async (
+  args: {
+    configPath: string
+    proxyAddress: string
+    silent: boolean
+  },
+  hre
+) => {
+  const { configPath, proxyAddress, silent } = args
+  const provider = hre.ethers.provider
+  const signer = provider.getSigner()
+
+  const spinner = ora({ isSilent: silent })
+  spinner.start('Checking project registration...')
+
+  const parsedConfig = loadParsedChugSplashConfig(configPath)
+
+  // Throw an error if the project has not been registered
+  if (
+    (await isProjectRegistered(signer, parsedConfig.options.projectName)) ===
+    false
+  ) {
+    errorProjectNotRegistered(
+      await getChainId(hre.ethers.provider),
+      hre.network.name,
+      configPath
+    )
+  }
+
+  spinner.succeed('Process registration detected')
+  spinner.start('Checking proxy compatibility...')
+
+  const incompatibleProxyError = `ChugSplash does not support your proxy type.
+  Currently ChugSplash only supports proxies that implement EIP-1967 which yours does not appear to do.
+  If you believe this is a mistake, please reach out to the developers or open an issue on GitHub.`
+
+  // Fetch proxy bytecode and check if it contains the expected EIP-1967 function definitions
+  const iface = new ethers.utils.Interface([
+    'function implementation() external view returns (address)',
+    'function admin() returns (address)',
+    'function upgradeTo(address)',
+    'function changeAdmin(address newAdmin)',
+  ])
```

Comment:
Instead of hard-coding the interface here, you can just import `ProxyABI` from `@chugsplash/contracts` and do:
```
const iface = new ethers.utils.Interface(ProxyABI)
```